data/org.freedesktop.impl.portal.AppChooser.xml

```diff
@@ -108,6 +112,10 @@
         This method can be called between the time of a ChooseApplication call
         and receiving the Response signal, to update the list of applications
         that are offered by the backend.
+
+        Note: To be able to handle legacy system apps the ids in `choices` are
+        in practice the base name of the desktop file without the `.desktop`
+        extension.
```

Comment:
@GeorgesStavracas we are not trying to lower the requirement, we are just clarifying the meaning, so that future backend implementation authors don't end up in the same confusion loop.

The changes made explicitly says that it is for legacy apps. Newly made apps and apps already following the standard are in no way affected too.

If you think we can reword it better so that it is clear, we can use those words instead :)

Reply:
@GeorgesStavracas The change documents the current practice the portal implementations follow. The alternative would be to have open-uri reject all non valid app-ids in the first place to have the result not being dependent on the backend implementation. Currently e.g. the gtk and gnome portals happily list vim/emacs for their given mime types although the desktop file is `vim.desktop`.